### PR TITLE
[CAI-90] New Chatbot Input Textarea

### PR DESCRIPTION
**Pull request by Sebastiano-Bertolin**
#### List of Changes
Update ChatInputText component with new style and behaviour

#### Motivation and Context
Update style as per mockup, implement the multi line functionality and max characters permitted

#### How Has This Been Tested?
Manually via Storybook

#### Screenshots (if appropriate):
Screenshot 2024-07-31 at 21 26 51
Screenshot 2024-07-31 at 21 27 22

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 1]
```diff
@@ -28,32 +29,53 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
...
         value={message}
         onChange={handleChange}
-        sx={{ ml: 1 }}
+        multiline
+        maxRows={8}
+        endAdornment={
+          <span style={{ color: alpha(palette.text.primary, 0.4) }}>
+            {message.length}/800
+          </span>
+        }
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:56]: image
Char counter should be align to bottom of text field and text size should be 12px in mobile and 14px in dekstop view

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx (hunk 1 above)
Comment (Sebastiano-Bertolin) [reply apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx]: Done

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 2]
```diff
@@ -12,9 +20,11 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
   const t = useTranslations();
   const [message, setMessage] = useState('');
   const { palette } = useTheme();
+  const disabledColor = alpha(palette.text.primary, 0.3);
+  const isDesktop = useMediaQuery((theme: Theme) => theme.breakpoints.up('lg'));
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:24]:
```suggestion
```

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 3]
```diff
@@ -27,33 +37,72 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
...
 
+  const onEnterKeyDownSubmitForm = (event: KeyboardEvent<HTMLFormElement>) => {
+    console.log(event.key);
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      submit();
+    }
+  };
+
+  const iconButtonSize = isDesktop ? '1.8rem' : '1.65rem';
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:48]:
```suggestion
```

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 4]
```diff
@@ -27,33 +37,72 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
...
+        endAdornment={
+          <span
+            style={{
+              color: alpha(palette.text.primary, 0.4),
+              fontSize: isDesktop ? '1rem' : '0.875rem',
+              marginLeft: '0.5rem',
+            }}
+          >
+            {message.length}/800
+          </span>
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:80]:
```suggestion
          <Box
            sx={{
              color: alpha(palette.text.primary, 0.4),
              fontSize: { xs: '0.875rem', md: '1rem' },
              marginLeft: '0.5rem',
            }}
          >
            {message.length}/800
          </Box>
```

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 5]
```diff
@@ -27,33 +37,72 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
...
+        }
+        sx={{
+          alignItems: 'flex-end',
+          borderWidth: '2px',
+          padding: isDesktop ? 2 : 1.5,
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:85]:
```suggestion
          padding: { xs: 1.5, md: 2 },
```

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 6]
```diff
@@ -27,33 +37,72 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
...
+          borderRadius: 2,
+          borderStyle: 'solid',
+          borderColor: message.length ? palette.primary.main : disabledColor,
+          fontSize: isDesktop ? '1rem' : '0.875rem',
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:89]:
```suggestion
          fontSize: { xs: '0.875rem', md: '1rem' },
```

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 7]
```diff
@@ -27,33 +37,72 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
...
       <IconButton
         aria-label='send'
         onClick={submit}
         disabled={!message || sendDisabled}
-        sx={{ p: '10px', color: palette.grey[700], cursor: 'pointer' }}
+        sx={{
+          p: '10px',
+          color: palette.primary.main,
+          cursor: 'pointer',
+          marginBottom: isDesktop ? 0.85 : 0,
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:100]:
```suggestion
          marginBottom: { xs: 0, md: 0.85 },
```

File: apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx [hunk 8]
```diff
@@ -27,33 +37,72 @@ const ChatInputText = ({ onSubmit, sendDisabled }: ChatInputTextProps) => {
...
+        }}
       >
-        <SendOutlined />
+        <Send sx={{ height: iconButtonSize, width: iconButtonSize }} />
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/atoms/ChatInputText/ChatInputText.tsx:103]:
```suggestion
        <Send
          sx={{
            height: { xs: '1.65rem', md: '1.8rem' },
            width: { xs: '1.65rem', md: '1.8rem' },
          }}
        />
```

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Comment by MarcoPonchia** [comment]
Included in https://github.com/pagopa/developer-portal/pull/1046 that is already merged

**Pull request closed without merging**